### PR TITLE
Add getCacheId() and clearCache() method to Translator

### PR DESCRIPTION
**Pull request by MatthiasKuehneEllerhold**
See https://github.com/zendframework/zend-i18n/pull/63 for discussion.

### CODE REVIEW

File: src/Translator/Translator.php [hunk 1]
```diff
@@ -561,6 +561,34 @@ public function addRemoteTranslations($type, $textDomain = 'default')
         return $this;
     }
 
+    /**
+     * Get the cache identifier for a specific textDomain and locale.
+     *
+     * @param    string $textDomain
+     * @param    string $locale
```
Comment (malukenho) [inline src/Translator/Translator.php]: Too much space

File: src/Translator/Translator.php [hunk 2]
```diff
@@ -561,6 +561,34 @@ public function addRemoteTranslations($type, $textDomain = 'default')
...
+    public function getCacheId($textDomain, $locale)
+    {
+        return 'Zend_I18n_Translator_Messages_' . md5($textDomain . $locale);
+    }
+
+    /**
+     * Clears the cache for a specific textDomain and locale.
+     *
+     * @param $textDomain
+     * @param $locale
```
Comment (malukenho) [inline src/Translator/Translator.php]: Parameter types?

**Review by malukenho — CHANGES_REQUESTED** [review]
I’m missing tests for the introduced api

**Comment by MatthiasKuehneEllerhold** [comment]
Fixed the comments and added tests.

Is there any consensus in the ZF code style about using static methods via `$this` ? (See Unit-Tests `$this->assertXY`...

**Comment by weierophinney** [comment]
> Is there any consensus in the ZF code style about using static methods via $this ?

For unit tests, that's the style we've consistently done since, well, we began. There's an [open discussion in our forums covering exactly this](https://discourse.zendframework.com/t/this-assert-vs-self-assert/448), though with no consensus as of yet.

**Comment by weierophinney** [comment]
Thanks, @MatthiasKuehneEllerhold. Merged to develop for release with 2.9.0.

**Pull request closed without merging**